pkg/steps/run.go

```diff
@@ -105,18 +108,28 @@ type subtestReporter interface {
 	SubTests() []*junit.TestCase
 }
 
-func runStep(ctx context.Context, node *api.StepNode, out chan<- message) {
+func runStep(ctx context.Context, node *api.StepNode, out chan<- Message) {
 	start := time.Now()
 	err := node.Step.Run(ctx)
 	var additionalTests []*junit.TestCase
 	if reporter, ok := node.Step.(subtestReporter); ok {
 		additionalTests = reporter.SubTests()
 	}
 	duration := time.Since(start)
-	out <- message{
+	failed := err != nil
+
+	out <- Message{
 		node:            node,
 		duration:        duration,
 		err:             err,
 		additionalTests: additionalTests,
+		stepDetails: api.CIOperatorStepWithDependencies{
+			StepName:    node.Step.Name(),
+			Description: node.Step.Description(),
+			StartedAt:   &start,
+			FinishedAt:  func() *time.Time { start.Add(duration); return &start }(),
```

Comment:
What is the purpose of the lambda?

Reply:
Getting a pointer for the result of `start.Add(duration)` inline